tests/test_ip.py

```
@@ -0,0 +1,296 @@
+
+    def test_standard_private_ip(self):
+        assert is_single_ipv4("192.168.1.1") is True
+
+    def test_loopback(self):
+        assert is_single_ipv4("127.0.0.1") is True
+
+    def test_public_ip(self):
+        assert is_single_ipv4("8.8.8.8") is True
+
+    def test_all_zeros(self):
+        assert is_single_ipv4("0.0.0.0") is True
```

Ruff (0.15.7)

[error] Possible binding to all interfaces

(S104)